src/components/UploadSection.svelte

```diff
@@ -14,12 +21,54 @@
         }
     });
 
+    async function sendFiles() {
+        let postRequests: Array<PostInsertRequestRequest> = [];
+        let postResponse: PostInsertRequestResponse[];
+
+        for (let i: number = 0; i < files.length; i++) {
+            let exifData = null;
+
+            try {
+                let exifData = await exifr.parse(files[i]);
+            } catch (e) {
+                console.error(e);
+            }
+
+            let takenAtSec = null;
+
+            if (exifData) {
+                takenAtSec = exifData.DateTimeOriginal.valueOf();
+            }
+
+            let postRequest: PostInsertRequestRequest = {
+                ObjectKey: files[i].name,
+                SizeBytes: files[i].size,
+                TakenAtSec: takenAtSec,
+            };
+
+            postRequests.push(postRequest);
+        }
+        try {
+            postResponse = await postInsertRequest(postRequests);
+            const loopTest = [];
+            for (let i = 0; i < files.length; i++) {
+                loopTest.push(
+                    pushMediaFilesToS3(postResponse[i].UploadUrl, files[i])
+                );
+            }
+            await Promise.all(loopTest);
+        } catch (e) {
+            console.error(e);
+        }
+    }
+
     function onFileSelected(e: Event) {
         if (e.target instanceof HTMLInputElement) {
             let image = Array.from(e.target.files);
             files = image;
             console.log(files);
         }
+        sendFiles();
```

Comment:
Do we need to use await here? I think we don't, because we don't need return result. Let's read about when we need to use await.

Reply:
Ok, i will read about it, but aggry with u, await is unpacking promise so we dont have returned value so we dont need to await